src/store/reducers/loadStateReducer.ts

```diff
@@ -0,0 +1,62 @@
+import { reducerDomains, loadStateActionTypes } from '../actions';
+
+const initialState = {
+    loadState: loadStateActionTypes.LOADED,
+    loadStatus: {
```

Comment:
so every reducer that uses load state will need to be added into the state here?

Reply:
So it seems like you're assuming that only one thing will ever be using this reducer at a time.  I think it is the wrong assumption.

Have you tested this loadStateReducer on different pages yet? I would have assumed it is breaking a lot of the UI.

Here's what I think will happen:
Even if you assume that only one network call will be made at a time (which is a very bold assumption), as long as you have another component on the same page that uses loadStateReducer and uses a different "reducerDomain", when you set error for one of the reducerDomains it will set error for all the reducerDomains on the page.

There are some ways to implement it so that it won't, like checking the loadStatus for every component, or doing some instantiation of a class (only applies to OOP). But it will be much more verbose, or increase complexity.